Sources/DiscordKitCore/Gateway/RobustWebSocket.swift

```diff
@@ -45,8 +62,22 @@ public class RobustWebSocket: NSObject {
     /// be attempted if/when this happens.
     public let onSessionInvalid = EventDispatch<Void>()
 
-    private var session: URLSession!, socket: URLSessionWebSocketTask!, decompressor: DecompressionEngine!
+    private var session: URLSession!, decompressor: DecompressionEngine!
+
+    #if canImport(WebSocket)
+    // We're using a 3rd party library on Linux because URLSessionWebSocketTask uses libcurl for its web socket communication.
+    // However, libcurl's websocket implementation is currently experimental, and you must build and install curl from source to use it.
+    // Because that is really bad UX, I elected to include an extremely hefty library to do our websocket stuff.
```

Comment:
Is it possible to statically-link a prebuilt and supported `libcurl` archive?

> It absolutely kills compile times, and inflates binary size unneccessarlly.

Sounds like this alternative also leads to really bad DX 😆

Reply:
> Is it possible to statically-link a prebuilt and supported libcurl archive?

Maybe, but since it would have to statically link to FoundationNetworking, I wouldn't want to do that. (Also I couldn't get it to work using dynamic linking, and I ran out of ideas for getting it to work.)

> Sounds like this alternative also leads to really bad DX

It's only really a problem for the first compile, subsequent compiles will be cached. The reason it is so big and takes so long to compile is because it has a dependency on swiftNIO. We may be able to provide a pre-built binary for that, but I'm not sure if it's possible.